x-pack/plugins/observability_solution/synthetics/server/alert_rules/common.ts

```diff
@@ -199,15 +202,14 @@ export const setRecoveredAlertsContext = ({
     let monitorSummary: MonitorSummaryStatusRule | null = null;
     let lastErrorMessage;
 
-    if (state?.idWithLocation && staleDownConfigs[state.idWithLocation]) {
-      const { idWithLocation, locationId } = state;
-      const downConfig = staleDownConfigs[idWithLocation];
-      const { ping, configId } = downConfig;
+    if (recoveredAlertId && locationId && staleDownConfigs[recoveredAlertId]) {
```

Comment:
Each monitor configuration has an id, and each location has an id. A single "monitor" is the combination of the monitor id and location id. This combination is what we use as the alert instance id.

We previously had the `idWithLocation` variable on state, but it actually matches the `recoveryAlertId` because that value was the same value we used as the alert instance id within the executor.